Gem/Code/Source/Components/UI/MatchOverComponent.cpp

```diff
@@ -0,0 +1,118 @@
+
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+#include <LyShine/Bus/UiCanvasBus.h>
+#include <LyShine/Bus/UiElementBus.h>
+
+namespace MultiplayerSample
+{
+    void MatchOverComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<MatchOverComponent, AZ::Component>()
+                ->Version(1)
+                ->Field("GameOverElementId", &MatchOverComponent::m_gameOverElementId)
+                ->Field("HudElementId", &MatchOverComponent::m_hudElementId)
+                ;
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<MatchOverComponent>("MatchOverComponent", "Handles game over display behavior")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "Multiplayer Sample UI")
+                    ->Attribute(AZ::Edit::Attributes::Icon, "Icons/Components/Component_Placeholder.svg")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &MatchOverComponent::m_gameOverElementId, "Game Over element ID", "The element ID of the Game Over UI element to display at Match end.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &MatchOverComponent::m_hudElementId, "HUD element ID", "The element ID of the UI element for the HUD")
+                    ;
+            }
+        }
+    }
+
+    void MatchOverComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC("NetworkMatchComponent"));
+        required.push_back(AZ_CRC("UiCanvasRefService"));
+    }
+
+    void MatchOverComponent::Activate()
+    {
+        UiCanvasRefBus::EventResult(m_uiCanvasId, GetEntityId(), &UiCanvasRefBus::Events::GetCanvas);
+        if (!m_uiCanvasId.IsValid())
+        {
+            UiCanvasAssetRefNotificationBus::Handler::BusConnect(GetEntityId());
+        }
+
+        if (NetworkMatchComponent* netMatchComponent = GetEntity()->FindComponent<NetworkMatchComponent>())
+        {
+            m_roundNumberHandler = AZ::EventHandler<uint16_t>([this](uint16_t value) 
+                { UpdateRound(value); });
+            netMatchComponent->RoundNumberAddEvent(m_roundNumberHandler);
+
+            m_roundTimerHandler = AZ::EventHandler<RoundTimeSec>([this](RoundTimeSec value) 
+                { DetermineIfMatchEnded(value); });
+            netMatchComponent->RoundTimeAddEvent(m_roundTimerHandler);
+        }
+    }
+
+    void MatchOverComponent::Deactivate()
+    {
+        UiCanvasAssetRefNotificationBus::Handler::BusDisconnect(GetEntityId());
```

Comment:
Tip: if you are shutting down you dont need to specify the id when calling BusDisconnect(), all buses can be called with an empty BusDisconnect().